### PR TITLE
Add missing files for iwslt18 recipe

### PR DESCRIPTION
**Pull request by hirofumi0810**
This PR is related to #707.

### CODE REVIEW

**Comment by hirofumi0810** [comment]
NOTE: this PR inclued a large dev/test split file for future research.

File: espnet/nets/pytorch_backend/decoders.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def forward(self, hs_pad, hlens, ys_pad, strm_idx=0):
             if i > 0 and random.random() < self.sampling_probability:
                 logging.info(' scheduled sampling ')
                 z_out = self.output(z_all[-1])
-                z_out = np.argmax(z_out.detach(), axis=1)
+                z_out = np.argmax(z_out.detach().cpu(), axis=1)
```
Comment (sw005320) [inline espnet/nets/pytorch_backend/decoders.py]: Do you need this?

File: espnet/asr/pytorch_backend/asr.py [hunk 2]
```diff
@@ -195,7 +195,9 @@ def __call__(self, batch, device):
         # perform padding and convert to tensor
         xs_pad = pad_list([torch.from_numpy(x).float() for x in xs], 0).to(device)
         ilens = torch.from_numpy(ilens).to(device)
-        ys_pad = pad_list([torch.from_numpy(y).long() for y in ys], self.ignore_id).to(device)
```
Comment (sw005320) [inline espnet/asr/pytorch_backend/asr.py]: We can keep the original one because the current speech translation recipe does not use multi-task learning.

File: egs/iwslt18st/st1/run.sh [hunk 3]
```diff
@@ -120,37 +132,81 @@ if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
     for x in train dev2010 tst2010 tst2013 tst2014 tst2015; do
         steps/make_fbank_pitch.sh --cmd "$train_cmd" --nj 32 --write_utt2num_frames true \
             data/${x} exp/make_fbank/${x} ${fbankdir}
-        utils/fix_data_dir.sh data/${x}
```
Comment (sw005320) [inline egs/iwslt18st/st1/run.sh]: Is it OK to remover this?

File: egs/iwslt18st/st1/run.sh [hunk 4]
```diff
@@ -55,24 +54,23 @@ maxlen_out=150 # if output length > maxlen_out, batchsize is automatically reduc
...
-penalty=0.2
-maxlenratio=0.8
-minlenratio=0
+penalty=0.3
+maxlenratio=1.6
+minlenratio=0.0
 recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
 
+# preprocessing related
+case=tc  # tc/lc/lc.rm
```
Comment (sw005320) [inline egs/iwslt18st/st1/run.sh]: maybe better to add more comments about what is tc, lc, and lc.rm.

File: egs/iwslt18st/st1/local/parse_xml.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python2
```
Comment (sw005320) [inline egs/iwslt18st/st1/local/parse_xml.py]: We don't recommend python2. It's OK for now, but I'll ask you to use python 3 later.

File: egs/iwslt18st/st1/local/forced_align.sh [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      cat ${trans}
+      echo ${duration}
+
+      # trim wav files
+      trim_wav=${tmpdir}/${utt_id}.wav
+      sox ${wav} ${trim_wav} trim ${start_time} ${duration} || exit 1;
+
+      # forced-align
+      log=${tmpdir}/${utt_id}.log
+      timeout 30 python3 /home/inaguma/tool/gentle/align.py ${trim_wav} ${trans} > ${log}
```
Comment (sw005320) [inline egs/iwslt18st/st1/local/forced_align.sh]: Please prepare `align.py`.

File: espnet/nets/pytorch_backend/decoders.py (hunk 1 above)
Comment (hirofumi0810) [reply espnet/nets/pytorch_backend/decoders.py]: Error occurred when using the original code in the current master branch.

File: espnet/asr/pytorch_backend/asr.py (hunk 2 above)
Comment (hirofumi0810) [reply espnet/asr/pytorch_backend/asr.py]: We always have two references per uttererance, one for source language (ASR) and another for target language (translation). I cannot run training scripts without this modification in the current master branch.

File: egs/iwslt18st/st1/run.sh (hunk 3 above)
Comment (hirofumi0810) [reply egs/iwslt18st/st1/run.sh]: It's OK.

File: egs/iwslt18st/st1/local/forced_align.sh (hunk 6 above)
Comment (hirofumi0810) [reply egs/iwslt18st/st1/local/forced_align.sh]: I removed this file because forced-alignment is not included in recipe.
`forced_align.sh` is a script to remove noisy utterances, but it is sufficient to include `reclist`.

**Comment by hirofumi0810** [comment]
@sw005320 All done. Please test in your environment.

File: tools/Makefile [hunk 7]
```diff
@@ -27,7 +27,7 @@ endif
 
 .PHONY: all clean
 
-all: kaldi.done python check_install
+all: kaldi.done python check_install extra
```
Comment (sw005320) [inline tools/Makefile]: It's better not to include `extra` to make the installation simple. What tools do you need to run?
Instead, you can add the installation check in `path.sh` see https://github.com/espnet/espnet/blob/d51251c799bd710e20804793e90c20f97cc6e60a/egs/csj/asr1/path.sh#L21-L33 as an example.

**Comment by sw005320** [comment]
Thanks! Let me try it then.

**Comment by hirofumi0810** [comment]
It modified scripts based on @sw005320 's comments.

**Comment by sw005320** [comment]
@gruly, if you have any comments, that is great. The recipe is now running at least in my environments. 

After one or two iterations, I'll merge this PR to the master first, and then merge to v.0.4.0.

**Comment by gruly** [comment]
@sw005320, thanks! We'll start trying it asap (maybe tonight if GPUs are available :-) )

File: espnet/asr/pytorch_backend/asr.py (hunk 2 above)
Comment (sw005320) [reply espnet/asr/pytorch_backend/asr.py]: I see.

**Comment by sw005320** [comment]
This is @hirofumi0810's great effort :)

**Comment by gruly** [comment]
Thanks @hirofumi0810!

**Comment by hirofumi0810** [comment]
@sw005320, It looks good for me now. One crutial issue is that it is difficult to train ST models from scratch in the current seq2seq model in espnet.
I know how to overcome this... use the context vector as the additonal feature before the final softmax for seqeunce generation in the decoder (called context residual in #563).
It will be included in #563. But should I include it here?

**Comment by sw005320** [comment]
> I know how to overcome this... use the context vector as the additonal feature before the final softmax for seqeunce generation in the decoder (called context residual in #563).
It will be included in #563. But should I include it here?

Let's do that with another PR.
Let me first check your recipe

**Comment by sw005320** [comment]
It seems that both recipes are working (at least in a training stage).

**Comment by sw005320** [comment]
@hirofumi0810, could you add the context vector modification in the v.0.4.0 branch?
